src/extension.ts

```diff
@@ -131,35 +135,102 @@ const toSeverity = (severity: ValeSeverity): DiagnosticSeverity => {
+    const range = new Range(
+        error.Line - 1,
+        error.Span[0] - 1,
+        error.Line - 1,
+        error.Span[1],
+    );
+    const message = error.Link
+        ? `${error.Message} (${error.Check}, see ${error.Link})`
+        : `${error.Message} (${error.Check})`;
     const diagnostic = new Diagnostic(
-        range, message, toSeverity(error.Severity));
+        range,
+        message,
+        toSeverity(error.Severity),
+    );
     diagnostic.source = "vale";
     diagnostic.code = error.Check;
     return diagnostic;
 };
 
-const readBinaryLocation = () => {
+/**
+ * Get the version of vale.
+ *
+ * @return A promise with the vale version string as single element.  If vale
+ * doesn't exist or if the version wasn't found the promise is rejected.
+ */
+const getValeVersion = async (
+    workspaceFolder?: WorkspaceFolder,
+): Promise<string> => {
+    const binaryLocation = readBinaryLocation(workspaceFolder);
+    // Run in an arbitrary directory, since "--version" doesn't depend on any
+    // folder
+    const stdout = await runInWorkspace(undefined, [
+        binaryLocation,
+        "--version",
+    ]);
+    const matches = stdout.match(/^vale version (.+)$/m);
+    if (matches && matches.length === 2) {
+        return matches[1];
+    }
+    throw new Error(`Failed to extract vale version from: ${stdout}`);
+};
+
+/**
+ * The version requirements for vale.
+ *
+ * We need at least 0.7.2 in this extension because earlier releases do not
+ * process all command line arguments, which we need to lint the entire
+ * workspace.
+ *
+ * See https://github.com/ValeLint/vale/issues/46 for details.
+ */
+const VERSION_REQUIREMENTS = ">= 0.7.2";
+
+const checkValeVersionSatisfies = async (workspaceFolder?: WorkspaceFolder) => {
+    const version = await getValeVersion(workspaceFolder);
+    if (
+        semver.satisfies(version, VERSION_REQUIREMENTS) ||
+        version === "master"
+    ) {
+        console.log(
+            `Found vale version ${version} satisfying ${VERSION_REQUIREMENTS}.`,
+        );
+    } else {
+        // tslint:disable-next-line:max-line-length
+        throw new Error(
+            `Vale version ${version} does not satisfy ${VERSION_REQUIREMENTS}`,
+        );
+    }
+};
+
+const readBinaryLocation = (workspaceFolder?: WorkspaceFolder) => {
     const configuration = workspace.getConfiguration();
-    const binaryPathRelativeToWorkspace =  configuration.get<string>('vscode-vale.path', 'vale');
-    return path.join(workspace.rootPath || '', binaryPathRelativeToWorkspace);
-}
+    const customBinaryPath = configuration.get<string>("vscode-vale.path");
+    if (customBinaryPath && workspaceFolder) {
+        return path.normalize(
+            customBinaryPath.replace(
+                "${workspaceFolder}",
+                workspaceFolder.uri.fsPath,
+            ),
+        );
+    }
+    // Assume that the binary is installed globally
+    return which.sync("vale", { pathExt: ".cmd" });
```

Comment:
In case no config path is specified, we assume that the binary is "globally" available in the `$PATH`.
The use of `which` (flow also does that) I suppose is mostly for windows support